portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +39,78 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+        int num = Integer.parseInt(request.getParameter("num"));
+        String post = request.getParameter("post");
+
+        Filter by_post = new FilterPredicate("post",FilterOperator.EQUAL,post);
+
+        Query query = new Query("Task").addSort("timestamp", SortDirection.DESCENDING);
+
+        query.setFilter(by_post);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Task> tasks = new ArrayList<>();
+        int i = 0;
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String text = (String) entity.getProperty("text");
+            long timestamp = (long) entity.getProperty("timestamp");
```

Comment:
Does the cast work?

Reply:
I believe so. Were you referring to the timestamp, text, or id?

Reply:
Timestamp. I thought Entity properties are strings but now I see it's actually Object.